src/torrent.cpp

```diff
@@ -9227,22 +9227,31 @@ namespace libtorrent
 
 	void torrent::queue_up()
 	{
+		// fix async position change calls for race conditions on changed torrents
+		if(!m_auto_managed || m_abort || is_finished()) return;
```

Comment:
can you explain this? where's the race condition? this is a function on ``torrent`` which is run in the libtorrent thread. there's no opportunity for the torrent to change state between the call to ``queue_position()`` and ``set_queue_position()``